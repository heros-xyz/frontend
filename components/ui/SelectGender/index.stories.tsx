import { HStack } from "@chakra-ui/react";
import type { Meta, StoryObj } from "@storybook/react";
import SelectGender from ".";

const meta: Meta<typeof SelectGender> = {
  title: "Components/SelectGender",
  component: SelectGender,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;

type Story = StoryObj<typeof SelectGender>;

export const SelectGenderComponent: Story = {
  render: (args) => (
    <HStack bg="primary" align={"center"} justify="center" padding={10}>
      <SelectGender {...args} />
    </HStack>
  ),
};
