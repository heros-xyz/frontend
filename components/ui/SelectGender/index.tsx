import { useEffect, useState } from "react";
import { Box, Flex, Text } from "@chakra-ui/react";
import { If, Then } from "react-if";
interface IProp {
  onChange: (value: string) => void;
  value: string;
  errorMessage?: string;
  flexRow?: boolean;
  bgColor?: string;
}

const listGender = [
  { label: "Male", value: "0" },
  { label: "Female", value: "1" },
  { label: "Other", value: "2" },
];

const SelectGender: React.FC<IProp> = ({
  value,
  errorMessage,
  onChange,
  flexRow,
  bgColor,
}) => {
  const [genderValue, setGenderValue] = useState(value);

  const isActive = (value: string) => {
    return genderValue === value;
  };

  useEffect(() => {
    onChange(genderValue);
  }, [genderValue]);
  const selectColor = bgColor === "primary" ? "accent.2" : "accent.2";
  return (
    <>
      <Flex
        flexDirection={flexRow ? "row" : { base: "column", lg: "row" }}
        gap={3}
      >
        {listGender.map((item, index) => (
          <Box
            key={`${"item" + index}`}
            w="100%"
            px={5}
            py={2}
            rounded="md"
            cursor="pointer"
            _hover={{
              color: selectColor,
            }}
            borderWidth={{
              base: isActive(item.value) ? "2px" : "1px",
              lg: "2px",
            }}
            borderColor={isActive(item.value) ? selectColor : "grey.100"}
            onClick={() => setGenderValue(item.value)}
          >
            <Text
              fontSize={
                flexRow === true
                  ? { base: "sm", xl: "lg" }
                  : { base: "sm", lg: "3xl" }
              }
              fontWeight={isActive(item.value) ? 700 : 500}
              color={isActive(item.value) ? selectColor : "grey.100"}
            >
              {item.label}
            </Text>
          </Box>
        ))}
      </Flex>
      <If condition={errorMessage}>
        <Then>
          <Text fontSize={{ base: "xs", xl: "sm" }} color="error.dark" mt={2}>
            This is a required field
          </Text>
        </Then>
      </If>
    </>
  );
};

export default SelectGender;
