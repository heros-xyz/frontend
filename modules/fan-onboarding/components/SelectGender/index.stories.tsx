import { Box } from "@chakra-ui/react";
import type { Meta, StoryObj } from "@storybook/react";
import SelectGender from "./";

const meta: Meta<typeof SelectGender> = {
  title: "Fan/On Boarding/SelectGender",
  component: SelectGender,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;
type Story = StoryObj<typeof SelectGender>;

export const SelectGenderComponent: Story = {
  render: (args) => (
    <Box h="100vh" bg="secondary">
      <SelectGender {...args} />
    </Box>
  ),
};
