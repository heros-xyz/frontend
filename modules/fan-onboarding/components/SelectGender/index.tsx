import { useState } from "react";
import { Box, Text } from "@chakra-ui/react";
import { ArrowRight } from "@/components/svg/ArrowRight";
import { GenderFanOnBoarding } from "@/components/svg/GenderFanOnBoarding";
import SelectGender from "@/components/ui/SelectGender";
import HerosOnboardingWrapperNew from "@/components/ui/HerosOnboardingWrapperNew";
interface EnterGenderProps {
  onSubmit: (gender: string) => void;
  gender: string;
}

const EnterGender: React.FC<EnterGenderProps> = ({ gender, onSubmit }) => {
  const [submitted, setSubmitted] = useState(false);
  const [genderValue, setgenderValue] = useState(gender);

  const handleSubmit = () => {
    setSubmitted(true);
    if (genderValue) {
      onSubmit(genderValue);
    }
  };

  return (
    <HerosOnboardingWrapperNew
      Icon={
        <GenderFanOnBoarding
          w={{ base: "80px", xl: "128px" }}
          h={{ base: "79px", xl: "127px" }}
        />
      }
      textButton="Proceed"
      IconButton={<ArrowRight />}
      onSubmit={handleSubmit}
      bgIconColor="accent.2"
      display="flex"
      alignItems="center"
      isPaddingTop={false}
    >
      <Box mb={6} color="primary">
        <Box mb={3.5} fontSize={{ lg: "xl" }} fontWeight="bold">
          Select Gender
          <Text as="span" color="error.dark">
            {" "}
            *
          </Text>
        </Box>
        <SelectGender
          value={genderValue}
          onChange={setgenderValue}
          errorMessage={
            !genderValue && submitted ? "This is a required field" : ""
          }
        />
      </Box>
    </HerosOnboardingWrapperNew>
  );
};

export default EnterGender;
