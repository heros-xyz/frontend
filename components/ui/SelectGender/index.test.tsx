import { render } from "@testing-library/react";
import SelectGender from ".";

it("renders select gender unchanged", () => {
  const { container } = render(
    <SelectGender onChange={console.log} value="null" />
  );
  expect(container).toMatchSnapshot();
});
