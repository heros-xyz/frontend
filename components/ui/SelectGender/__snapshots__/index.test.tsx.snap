// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders select gender unchanged 1`] = `
<div>
  <div
    class="css-1omp9s3"
  >
    <div
      class="css-qh0ll2"
    >
      <p
        class="chakra-text css-j7hta7"
      >
        Male
      </p>
    </div>
    <div
      class="css-qh0ll2"
    >
      <p
        class="chakra-text css-j7hta7"
      >
        Female
      </p>
    </div>
    <div
      class="css-qh0ll2"
    >
      <p
        class="chakra-text css-j7hta7"
      >
        Other
      </p>
    </div>
  </div>
</div>
`;
